src/components/stats/statsQuerySrv.js

```diff
@@ -1,10 +1,96 @@
 (function (angular) {
     'use strict';
 
-    angular.module('znk.infra.stats').factory('StatsQuerySrv', [
-        function () {
+    angular.module('znk.infra.stats').service('StatsQuerySrv', [
+        'StatsSrv',
+        function (StatsSrv) {
             var StatsQuerySrv = {};
 
+            function _getCategoryWeakness(category) {
+                if (!category.totalQuestions) {
+                    return -Infinity;
```

Comment:
Why do we need to configure it ? it is javascript value....
yeah in case the category has no total questions (scenario which should not happen)  then it will be ignored.